test/unit/test_http_utils.py

```diff
@@ -125,26 +126,59 @@ class TestSomewhatBufferedHTTPConnection(unittest.TestCase):
             self, mock_http_response_init):
+        with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)) \
+                as sock:
 
-        kwargs = dict(
-            sock=None, debuglevel=0, strict=0, method=None, buffering=False)
+            kwargs = dict(
+                sock=sock, debuglevel=0, strict=0, method=None, buffering=False)
 
-        response_class = \
-            swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
-        response_obj = response_class(**kwargs)
+            response_class = \
+                swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
+            response_obj = response_class(**kwargs)
 
-        if sys.version_info < (2, 7):
-            del kwargs['buffering']
+            if sys.version_info < (2, 7):
+                del kwargs['buffering']
 
-        mock_http_response_init.assert_called_once_with(
-            response_obj, **kwargs)
+            mock_http_response_init.assert_called_once_with(
+                response_obj, **kwargs)
 
-    @mock.patch('httplib.HTTPResponse.__init__', mock.Mock())
     def test_fileno_attr(self):
-        response_class = \
-            swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
-        response_obj = response_class(sock=None)
         with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)) \
                 as sock:
-            response_obj.fp = sock
+            response_class = \
+                swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
+            response_obj = response_class(sock=sock)
             self.assertEquals(sock.fileno(), response_obj.fileno())
+
+    @mock.patch('socket._socketobject.makefile')
+    def test_makefile_not_called(
+            self, mock_makefile):
+        with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)) \
+                as sock:
+
+            kwargs = dict(
+                sock=sock, debuglevel=0, strict=0, method=None)
+
+            response_class = \
+                swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
+            response_class(**kwargs)
+
+            self.assertFalse(mock_makefile.called)
+
+    def test_non_keepalive_conn(self):
+
+        content = 'Hello, World!'
+
+        def my_app(env, start_response):
+            start_response('200 OK', [('Content-Type', 'text/plain'),
+                                      ('Connection', 'close')])
+            return [content]
+
+        with utils.WSGIServer(my_app) as server:
+            conn_class = \
+                swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection
+            conn = conn_class(server.ip, server.port)
+            conn.request("GET", "/")
+            conn_sock_fd = conn.sock.fileno()
+            response = conn.getresponse()
+            self.assertEqual(conn_sock_fd, response.fileno())
```

Comment:
It should not be. The original bug was detected because the socket was closed too early and therefore response.fileno() raise an exception. 
Here we want to test that SomewhatBufferedHTTPConnection and its HTTPResponse share the same socket and that it is not closed at that time.
This might raise another issue : when the socket should be closed then ?